src/blob_gc_job.cc

```diff
@@ -269,46 +269,46 @@ Status BlobGCJob::DoRunGC() {
   return s;
 }
 
-void BlobGCJob::BatchWriteNewIndices(BlobFileBuilder::OutContexts& contexts,
-                                     Status* s) {
+Status BlobGCJob::BatchWriteNewIndices(const BlobFileBuilder::OutContexts& contexts) {
   auto* cfh = blob_gc_->column_family_handle();
-  for (const std::unique_ptr<BlobFileBuilder::BlobRecordContext>& ctx :
-       contexts) {
-    MergeBlobIndex merge_blob_index;
-    merge_blob_index.file_number = ctx->new_blob_index.file_number;
-    merge_blob_index.source_file_number = ctx->original_blob_index.file_number;
-    merge_blob_index.source_file_offset =
-        ctx->original_blob_index.blob_handle.offset;
-    merge_blob_index.blob_handle = ctx->new_blob_index.blob_handle;
-
+  Status s;
+  for (const auto& ctx : contexts) {
     std::string index_entry;
     BlobIndex original_index = ctx->original_blob_index;
     ParsedInternalKey ikey;
     if (!ParseInternalKey(ctx->key, &ikey)) {
-      *s = Status::Corruption(Slice());
-      return;
+      return Status::Corruption(Slice());;
     }
     if (!gc_merge_rewrite_) {
-      merge_blob_index.EncodeToBase(&index_entry);
+      const BlobIndex &index = ctx->new_blob_index;
```

Comment:
why is it different from before?

Reply:
Because the merge blob index is only used in the second if branch. I just think we should only declare the merge blob index when we truly use it. This will help others understand the code easily.